OpenRA.Mods.Common/Traits/AppearsOnRadar.cs

```diff
@@ -40,7 +40,31 @@ public void Created(Actor self)
 
 		public IEnumerable<Pair<CPos, Color>> RadarSignatureCells(Actor self)
 		{
-			var color = modifier != null ? modifier.RadarColorOverride(self) : self.Owner.Color.RGB;
+			var player = self.World.RenderPlayer ?? self.World.LocalPlayer;
+			var color = self.Owner.Color.RGB;
+
+			if (modifier != null)
+				color = modifier.RadarColorOverride(self);
+
+			if (Game.Settings.Game.AllianceDisplay && player != null && !player.Spectating)
```

Comment:
@pchote where should I put this helper method so it can be shared across here and health bars?